src/main/java/org/openrewrite/java/jackson/codehaus/RemoveDoublyAnnotatedCodehausAnnotations.java

```diff
@@ -50,12 +54,33 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                     @Override
                     public J preVisit(@NonNull J tree, ExecutionContext ctx) {
                         stopAfterPreVisit();
-                        doAfterVisit(new RemoveAnnotationVisitor(new AnnotationMatcher("@org.codehaus.jackson.map.annotate.JsonSerialize", true)));
+
+                        Set<J.Annotation> annotationsToRemove = new FindDoublyAnnotatedVisitor().reduce(tree, new HashSet<>());
+                        AnnotationMatcher matcher = new AnnotationMatcher("@org.codehaus.jackson.map.annotate.JsonSerialize", true) {
+                            @Override
+                            public boolean matches(J.Annotation annotation) {
+                                return super.matches(annotation) && annotationsToRemove.contains(annotation);
```

Comment:
Nice and creative use here with minimal lines of code still; The only thing I'd been wondering if we can simplify is that we now appear to match on codehaus annotations twice; once to find duplicates, and once more when we call `super` here. That's perhaps not necessary, but I also don't quite like the immediate replacement of not calling super, as that discards the AnnotationMatcher constructor argument passed in. Any other suggestions?
```suggestion
                        Set<J.Annotation> annotationsToRemove = new FindDoublyAnnotatedVisitor().reduce(tree, new HashSet<>());
                        AnnotationMatcher matcher = new AnnotationMatcher("@org.codehaus.jackson.map.annotate.JsonSerialize", true) {
                            @Override
                            public boolean matches(J.Annotation annotation) {
                                return annotationsToRemove.contains(annotation);
```